recipes/ninja-hep-ph/meta.yaml

```diff
@@ -0,0 +1,143 @@
+{% set name = "ninja" %}
+{% set version = "1.2.0" %}
+
+package:
+  name: {{ name }}-hep-ph
+  version: {{ version }}
+
+source:
+  url: https://github.com/peraro/ninja/archive/v{{ version }}.tar.gz
+  sha256: b9709aada408286b6533ac7427bab9d1364d449a9c11098c324ea0875fe9a9e1
+  patches:
+    - posix-compliant-shell.patch
+    - update-version-number-to-v1.2.0.patch
+
+build:
+  # libquadmath is required, so macOS is not supported
+  skip: true  # [not linux]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('ninja-hep-ph', max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ stdlib('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - automake  # includes autoconf
+    - libtool
+    - make
+    - pkg-config
+    - gnuconfig
+    - oneloop  # [build_platform != target_platform]
+    - looptools-static  # [build_platform != target_platform]
+  host:
+    - oneloop
+    - looptools-static
+  run:
+    - oneloop
+    - looptools-static
```

Comment:
Why is a static library needed at runtime?

Reply:
@xhochy Here I'm going to have to plead ignorance as to why a shared vs. static library matters. For applications that use `libninja`, `libooptools` is going to be needed as well, and so it is included.

e.g. the example in the tests:

```
$CXX simple_test.cc mynum.cc -o simple_test $CXXFLAGS -I./ -I$PREFIX/include/ninja $LDFLAGS -lninja -lavh_olo -looptools -lgfortran -lquadmath
```

Though perhaps you're saying that run dependencies in "the real world" don't constitute **required** run dependencies to have anything work at all? And that these should just be added to the `test` `requires`?

Reply:
Do you actually need to link to `libooptools` for building the test? If you do that might point towards the need for a libooptools-abi package to track the ABI between downstream packages.

Regardless, `libooptools.a` won't actually be needed at runtime.

Here's something I got from an LLM to maybe clarify things while still being lazy :)

> The distinction between shared and static libraries does impact whether a dependency is required at runtime. In the case of ooptools, which is a static library, it is linked directly into the application or library at build time. This means that once the binary (e.g., `libninja` or `simple_test`) is built, ooptools is no longer needed at runtime because its code has already been included in the final binary.
>
> On the other hand, for ninja, which is a shared library, its presence is required at runtime because the application dynamically links to it. Therefore, it makes sense for ninja to be a runtime dependency, while ooptools can be considered only a build dependency.
>
> To your second point, if ooptools is only required during the build process (and not at runtime), it should indeed be listed under the test requires or the host dependencies in the Conda recipe. This helps to accurately reflect the distinction between build-time and runtime requirements and ensures that users aren't installing unnecessary dependencies when they only need to run the application.

Reply:
> Do you actually need to link to libooptools for building the test?

@chrisburr Yes, if I use the following diff

```diff
diff --git a/recipes/ninja-hep-ph/meta.yaml b/recipes/ninja-hep-ph/meta.yaml
index 885c7052c6..195bfab219 100644
--- a/recipes/ninja-hep-ph/meta.yaml
+++ b/recipes/ninja-hep-ph/meta.yaml
@@ -36,7 +36,6 @@ requirements:
     - looptools-static
   run:
     - oneloop
-    - looptools-static
 
 test:
   # As need to regenerate Makefiles, need all of the source files that
@@ -53,6 +52,7 @@ test:
     - automake
     - libtool
     - make
+    - looptools-static
   commands:
     - test -f $PREFIX/lib/libninja${SHLIB_EXT}
     # c.f. https://github.com/conda/conda-build/issues/3075
@@ -109,7 +109,7 @@ test:
     # Show how to build manually
     - make clean
     - echo -e "\n# simple_test by hand"
-    - $CXX simple_test.cc mynum.cc -o simple_test $CXXFLAGS -I./ -I$PREFIX/include/ninja $LDFLAGS -lninja -lavh_olo -looptools -lgfortran -lquadmath
+    - $CXX simple_test.cc mynum.cc -o simple_test $CXXFLAGS -I./ -I$PREFIX/include/ninja $LDFLAGS -lninja -lavh_olo -lgfortran -lquadmath
     - ./simple_test
 
     - make clean
```

then `pixi run build-linux linux64` fails at

```
./simple_test
```

with

```
./simple_test: symbol lookup error: /home/conda/staged-recipes/build_artifacts/ninja-hep-ph_1733470423450/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/libninja.so.0: undefined symbol: clearcache_
```

If I have my compilation understanding right, this is because `ninja`'s ["built-in interface to the LoopTools library"](https://github.com/peraro/ninja/blob/227b0747e2f5c9509966dc5526fa2f2e705a5041/README.md#installing-ninja) needs to link the library's symbols for the interface to work as `libooptools.a` isn't getting built into any other library.

> Regardless, `libooptools.a` won't actually be needed at runtime.

I think my lack of C++ skill is showing here, but if a library is required for an interface to work at execution time, isn't that the same as a runtime requirement?

Your LLM's summary

> This means that once the binary (e.g., `libninja` or `simple_test`) is built, `ooptools` is no longer needed at runtime because its code has already been included in the final binary.

doesn't seem correct just given the execution failure.

Maybe I'm doing something incredibly dumb, but I feel like if nothing else about my (limited) knowledge of linkers,  all of our HEP tools `Xtool-config --ldflags` have taught me that what I'm doing is reasonable.

Reply:
@chrisburr I should note that I don't _really_ know how to use `ninja`, I'm just trying to get it packaged so that we can eventually package  `mg5amcnlo` (https://github.com/hep-packaging-coordination/packaging-hep-simulation-stack/issues/3). So for how `ninja`'s `libooptools` interface works we might need to tag in @peraro again, as they designed it.

Reply:
> Regardless, `libooptools.a` won't actually be needed at runtime.

Oh. I think I understand what you mean now. It is required at **linktime** not runtime. So I can compile and link to `libooptools.a` but then delete the library and the executable will still run fine.

```diff
diff --git a/recipes/ninja-hep-ph/meta.yaml b/recipes/ninja-hep-ph/meta.yaml
index 885c7052c6..1ecfad890d 100644
--- a/recipes/ninja-hep-ph/meta.yaml
+++ b/recipes/ninja-hep-ph/meta.yaml
@@ -36,7 +36,6 @@ requirements:
     - looptools-static
   run:
     - oneloop
-    - looptools-static
 
 test:
   # As need to regenerate Makefiles, need all of the source files that
@@ -53,6 +52,7 @@ test:
     - automake
     - libtool
     - make
+    - looptools-static
   commands:
     - test -f $PREFIX/lib/libninja${SHLIB_EXT}
     # c.f. https://github.com/conda/conda-build/issues/3075
@@ -110,6 +110,9 @@ test:
     - make clean
     - echo -e "\n# simple_test by hand"
     - $CXX simple_test.cc mynum.cc -o simple_test $CXXFLAGS -I./ -I$PREFIX/include/ninja $LDFLAGS -lninja -lavh_olo -looptools -lgfortran -lquadmath
+    - test -f $PREFIX/lib/libooptools.a
+    - rm -f $PREFIX/lib/libooptools.a
+    - test ! -f $PREFIX/lib/libooptools.a
     - ./simple_test
 
     - make clean
```

Okay yes, I was being dumb. Thank you both.

Reply:
Thanks for being very responsive and helpful, @peraro!

Things are working now, and others might benefit from having the interface, so I'm happy to maintain it.